### ISSUE TITLE
Memory leak on 1.6.14

### ISSUE THREAD

**Issue by fauzanardh** [lucidrains/vector-quantize-pytorch#56]
I'm encountering a memory leak after updating to 1.6.14 from 1.6.11.
At first, I noticed that my training code, which was previously working just fine, is now encountering an OOM error after several iterations, which is weird.
I can confirm the memory leak issue come from this library, because when I downgraded the library back to 1.6.11, the OOM error were no longer present.

The model follows this structure:
```python
self.encoder = Encoder(...)
self.vq = GroupedResidualVQ(
    dim=dim_emb,
    num_quantizers=num_codebooks,
    codebook_size=n_emb,
    groups=vq_groups,
    decay=vq_decay,
    commitment_weight=vq_commitment_weight,
    quantize_dropout_multiple_of=1,
    quantize_dropout=True,
    quantize_dropout_cutoff_index=vq_quantize_dropout_cutoff_index,
    kmeans_init=True,
    threshold_ema_dead_code=2,
    stochastic_sample_codes=vq_stochastic_sample_codes,
)
self.decoder = Decoder(...)
```

And the snippet of the training code is like this:
```python
x = model.encoder(x)
quantized, _, commit_loss = model.vq(x)
recon_x = model.decoder(quantized)
```

For context, I trained this on a Google Colab environment.

**Comment by lucidrains** [comment]
@fauzanardh my bad, do you want to try 1.6.15?

**Comment by fauzanardh** [comment]
Unfortunately, the issue still persist on 1.6.15

**Comment by lucidrains** [comment]
@fauzanardh hmm, could you try turning off the commitment loss `commitment_weight = 0` (that would help me narrow it down if that resolves it)

also give 1.6.16 a quick try too

**Comment by fauzanardh** [comment]
Both upgrading to 1.6.16 and setting the `commitment_weight` to 0 still results in OOM, could it be because of this commit 8a5a1592c68213e59dc409d738f340982693c3a3?

**Comment by lucidrains** [comment]
@fauzanardh yea it could be, let's try 1.6.17, and if that doesn't work, just revert that commit

**Comment by fauzanardh** [comment]
1.6.17 also doesn't seem to fix it.
I tried to find the last GitHub commit that works, and found that this commit ae5db0169772ae4c624e65d3b5c33f45cdd34a92 is the last commit that doesn't spew out the OOM error. So I think that the problem might be coming from this commit 4a448199b6fb52c616095d0023bee700d219c739, which is the first commit that produced OOM error. And not this one 8a5a1592c68213e59dc409d738f340982693c3a3, so my last point was probably wrong.

**Comment by lucidrains** [comment]
@fauzanardh thanks for identifying the commit! let's try 1.6.18?

**Comment by fauzanardh** [comment]
Yea, that commit fixed it 🎉!
I was about to comment about that whether it is intentional or not, because it seemed weird that you only passed the `self.cluster_size` and not the `self.cluster_size.data`

**Comment by lucidrains** [comment]
@fauzanardh yes indeed

i blame all the traveling and the jet lag lol
